Instantly crash upon opening the app, from what I can tell it looks like the code does not consider the case inwhich the token is an empty string... see error from android below

Caused by: java.lang.NullPointerException
            at edu.uco.schambers.classmate.Database.TokenUtility.parseUserToken(TokenUtility.java:11)
            at edu.uco.schambers.classmate.Fragments.Auth.initUI(Auth.java:141)
            at edu.uco.schambers.classmate.Fragments.Auth.onCreateView(Auth.java:100)
            at android.app.Fragment.performCreateView(Fragment.java:1699)

--- a/app/src/main/java/edu/uco/schambers/classmate/Activites/MainActivity.java
+++ b/app/src/main/java/edu/uco/schambers/classmate/Activites/MainActivity.java
@@ -138,8 +138,6 @@ public class MainActivity extends Activity implements StudentResponseFragment.On
             sp = this.getSharedPreferences(MyPREFS, Context.MODE_PRIVATE);
             editor = sp.edit();
             editor.remove("AUTH_TOKEN");
-            editor.clear();
-            editor.putString("AUTH_TOKEN", null);
             editor.commit();
 
             FragmentTransaction trans = getFragmentManager().beginTransaction();

--- a/app/src/main/java/edu/uco/schambers/classmate/Activites/MainActivity.java
+++ b/app/src/main/java/edu/uco/schambers/classmate/Activites/MainActivity.java
@@ -137,13 +137,9 @@ public class MainActivity extends Activity implements StudentResponseFragment.On
         {
             sp = this.getSharedPreferences(MyPREFS, Context.MODE_PRIVATE);
             editor = sp.edit();
-            Log.i("before remove", sp.getString("AUTH_TOKEN", null));
             editor.remove("AUTH_TOKEN");
-            Log.i("after remove", sp.getString("AUTH_TOKEN", null));
             editor.clear();
-            Log.i("after clear", sp.getString("AUTH_TOKEN", null));
             editor.putString("AUTH_TOKEN", null);
-            Log.i("after putString", sp.getString("AUTH_TOKEN", null));
             editor.commit();
 
             FragmentTransaction trans = getFragmentManager().beginTransaction();

--- a/app/src/main/java/edu/uco/schambers/classmate/Activites/MainActivity.java
+++ b/app/src/main/java/edu/uco/schambers/classmate/Activites/MainActivity.java
@@ -137,9 +137,13 @@ public class MainActivity extends Activity implements StudentResponseFragment.On
         {
             sp = this.getSharedPreferences(MyPREFS, Context.MODE_PRIVATE);
             editor = sp.edit();
+            Log.i("before remove", sp.getString("AUTH_TOKEN", null));
             editor.remove("AUTH_TOKEN");
+            Log.i("after remove", sp.getString("AUTH_TOKEN", null));
             editor.clear();
-            editor.putString("AUTH_TOKEN", "");
+            Log.i("after clear", sp.getString("AUTH_TOKEN", null));
+            editor.putString("AUTH_TOKEN", null);
+            Log.i("after putString", sp.getString("AUTH_TOKEN", null));
             editor.commit();
 
             FragmentTransaction trans = getFragmentManager().beginTransaction();

--- a/app/src/main/java/edu/uco/schambers/classmate/Fragments/Auth.java
+++ b/app/src/main/java/edu/uco/schambers/classmate/Fragments/Auth.java
@@ -133,6 +133,18 @@ public class Auth extends Fragment {
         signup = (TextView)  rootView.findViewById(R.id.signup_lbl);
         resetLink = (TextView) rootView.findViewById(R.id.reset_pw_lbl);
         dr = new DataRepo(getActivity());
+        sp = getActivity().getSharedPreferences(MyPREFS, Context.MODE_PRIVATE);
+
+        if(sp.contains("AUTH_TOKEN")) {
+            token = sp.getString("AUTH_TOKEN", null);
+            try {
+                User user = TokenUtility.parseUserToken(token);
+                ChooseInterface(user.isStudent());
+            } catch (JSONException e) {
+
+            }
+        }
+
 
         email.setOnFocusChangeListener(new View.OnFocusChangeListener() {
             @Override

